clusterloader2/pkg/measurement/util/prometheus.go

```diff
@@ -74,6 +88,73 @@ type promResponseData struct {
 	v model.Value
 }
 
+// ExecutePrometheusQuery executes given promethous query.
+// If not config is provided the default one is used.
+func ExecutePrometheusQuery(c clientset.Interface, query string, queryTime time.Time, config *PrometheusConfig) ([]*model.Sample, error) {
+	if queryTime.IsZero() {
+		return nil, fmt.Errorf("query time can't be zero")
+	}
+
+	if config == nil {
+		config = &PrometheusConfig{
+			Namespace: defaultNamespace,
+			Scheme:    defaultScheme,
+			Name:      defaultName,
+			Port:      defaultPort,
+			Path:      defaultPath,
+		}
+	}
+
+	var body []byte
+	var queryErr error
+	params := map[string]string{
+		"query": query,
+		"time":  queryTime.Format(time.RFC3339),
+	}
+	if err := wait.PollImmediate(queryInterval, queryTimeout, func() (bool, error) {
+		body, queryErr = c.CoreV1().
+			Services(config.Namespace).
+			ProxyGet(config.Scheme, config.Name, config.Port, config.Path, params).
+			DoRaw()
+		if queryErr != nil {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		if queryErr != nil {
+			return nil, fmt.Errorf("query error: %v", queryErr)
+		}
+		return nil, fmt.Errorf("query error: %v", err)
+	}
+
+	samples, err := ExtractMetricSamples2(body)
+	if err != nil {
+		return nil, fmt.Errorf("exctracting error: %v", err)
+	}
+
+	var resultSamples []*model.Sample
+	for _, sample := range samples {
+		if !math.IsNaN(float64(sample.Value)) {
+			resultSamples = append(resultSamples, sample)
+		}
+	}
+	return resultSamples, nil
+}
+
+// PrometheusConfig contains data required to access promethous server.
+type PrometheusConfig struct {
```

Comment:
TBH, I don't see anything here that clients of this util would like to customize in the near future. IMHO we can hide all these setting inside the implementation for now.

Reply:
+1

Reply:
Removed.